webpack.config.js

```diff
@@ -0,0 +1,182 @@
+const cssnano = require('cssnano');
+const CopyPlugin = require('copy-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const sass = require('sass');
+const ESLintPlugin = require('eslint-webpack-plugin');
+const StylelintPlugin = require('stylelint-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+
+const projectRoot = 'wagtailio';
+
+const options = {
+  entry: {
+    main: `./${projectRoot}/static/js/main.js`,
+  },
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js'],
+  },
+  output: {
+    path: path.resolve(`./${projectRoot}/static_compiled/`),
+    filename: 'js/[name].js',
+  },
+  plugins: [
+    new CopyPlugin({
+      patterns: [
+        {
+          from: 'img',
+          context: path.resolve(`./${projectRoot}/static/`),
+          to: path.resolve(`./${projectRoot}/static_compiled/img`),
+        },
+      ],
+    }),
+    new MiniCssExtractPlugin({
+      filename: 'css/[name].css',
+    }),
+    new ESLintPlugin({
+      failOnError: false,
+      lintDirtyModulesOnly: true,
+      emitWarning: true,
+    }),
+    new StylelintPlugin({
+      failOnError: false,
+      lintDirtyModulesOnly: true,
+      emitWarning: true,
+      extensions: ['scss'],
+    }),
+    //  Automatically remove all unused webpack assets on rebuild
+    new CleanWebpackPlugin(),
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.(js|ts|tsx)$/,
+        exclude: /node_modules/,
+        use: {
+          loader: 'ts-loader',
+        },
+      },
+      {
+        test: /\.(scss|css)$/,
+        use: [
+          MiniCssExtractPlugin.loader,
+          {
+            loader: 'css-loader',
+            options: {
+              sourceMap: true,
+            },
+          },
+          {
+            loader: 'postcss-loader',
+            options: {
+              postcssOptions: {
+                sourceMap: true,
+                plugins: () => [
+                  autoprefixer(),
+                  cssnano({
+                    preset: 'default',
+                  }),
+                ],
+              },
+            },
+          },
+          {
+            loader: 'sass-loader',
+            options: {
+              sourceMap: true,
+              implementation: sass,
+              sassOptions: {
+                outputStyle: 'compressed',
+              },
+            },
+          },
+        ],
+      },
+      {
+        // sync font files referenced by the css to the fonts directory
+        // the publicPath matches the path from the compiled css to the font file
+        // only looks in the fonts folder so pngs in the images folder won't get put in the fonts folder
+        test: /\.(woff|woff2)$/,
+        include: /fonts/,
+        use: {
+          loader: 'file-loader',
+          options: {
+            name: '[name].[ext]',
+            outputPath: 'fonts/',
+            publicPath: '../fonts',
+          },
+        },
+      },
```

Comment:
We use a system font for Wagtail so I believe you shouldn’t need this, nor `file-loader` as a dependency.

Reply:
Thanks. Removed this and `file-loader`